tools/utility/messageFactory.py

```diff
@@ -0,0 +1,108 @@
+
+        LOGGER.info('Initialising Message')
+        config_path = os.path.abspath('.') + '/' + config_path
+        config = ConfigParser.ConfigParser()
+        config.read(config_path)
+        token = config.get('Message-Section', 'access-token')
+        self.__api = CiscoSparkAPI(access_token=token)
+        rooms = list_matching_rooms(self.__api, 'YANG Catalog admin')
+
+        if len(rooms) == 0:
+            LOGGER.error('Need at least one room')
+            sys.exit(1)
+        if len(rooms) != 1:
+            print('Too many rooms! Refine the name:')
+            for r in rooms:
+                LOGGER.info('{}'.format(r.title))
+            sys.exit(1)
+
+        # Ok, we should have just one room if we get here
+        self.__room = rooms[0]
+        self.__smtp = smtplib.SMTP('localhost')
+
+    def __post_to_spark(self, msg, markdown=False):
+        if markdown:
+            self.__api.messages.create(self.__room.id, markdown=msg)
+        else:
+            self.__api.messages.create(self.__room.id, text=msg)
+
+    def __post_to_email(self, message, to):
+        if not isinstance(to, list):
+            to = [to]
+        msg = MIMEText(message)
+        msg['Subject'] = 'Automatic generated message - RFC ietf'
+        msg['From'] = 'no-reply@yangcatalog.org'
+        msg['To'] = ', '.join(to)
+
+        self.__smtp.sendmail('no-reply@yangcatalog.org', to, msg.as_string())
+        self.__smtp.quit()
+
+    def send_new_rfc_message(self, new_files, diff_files):
+        LOGGER.info('Sending notification about new IETF RFC modules')
+        new_files = '\n'.join(new_files)
+        diff_files = '\n'.join(diff_files)
+        message = ('{}\n\nSome of the files are different'
+                   ' in http://www.claise.be/IETFYANGRFC.json against'
+                   ' yangModels/yang repository\n\n'
+                   'Files that are missing in yangModles repo: \n{} \n\n '
+                   'Files that are different then in yangModels repo: \n{}'
+                   .format(GREETINGS, new_files, diff_files))
+        to = ['bclaise@cisco.com', 'einarnn@cisco.com', 'jclarke@cisco.com']
+
+        self.__post_to_spark(message)
+        self.__post_to_email(message, to)
+
+    def send_travis_auth_failed(self):
+        LOGGER.info('Sending notification about travis authorization failed')
+        message = ('Travis pull job not sent because patch was not sent from'
+                   ' travis. Key verification failed')
+        self.__post_to_spark(message)
+
+    def send_removed_temp_diff_files(self):
+        # TODO send spark message about removed searched diff files
+        pass
+
+    def send_removed_yang_files(self, removed_yang_files):
+        LOGGER.info('Sending notification about removed yang modules')
+        message = ('The following files has been removed from yangcatalog.org'
+                   ' using api: \'\'\'{}\'\'\''.format(removed_yang_files))
```

Comment:
You use the pattern \’\’\’ in a number of places. Not sure why. You could also do something like:

```
print(“‘’’{}’’’”.format(blah))
```

...which looks neater.

Or did you really mean to use backticks for markdown formatting?

Reply:
I did mean to use a backticks. it should be formatted code.

Reply:
Ok, let me know when you’ve updated to use backticks rather than escaped single quotes.

Reply:
Done